components/style/properties/longhand/font.mako.rs

```
@@ -361,3 +361,105 @@ ${helpers.single_keyword("font-variant-position",
+            }
+        }
+
+        impl FeatureTagValue {
+            /// <string> [ on | off | <integer> ]
+            pub fn parse(input: &mut Parser) -> Result<FeatureTagValue, ()> {
+                let tag = try!(input.expect_string()).into_owned();
+
+                // allowed strings of length 4 containing chars: <U+20, U+7E>
+                if tag.len() != 4 ||
+                   tag.chars().any(|c| c < ' ' || c > '~') {
+                       return Err(())
+                   }
+
+                if let Ok(value) = input.try(|input| input.expect_integer()) {
+                    // handle integer, throw if it is negative
+                    if value >= 0 {
+                        Ok(FeatureTagValue{ tag: tag, value: value })
+                    } else {
+                        Err(())
+                    }
+                } else if let Ok(_) = input.try(|input| input.expect_ident_matching("on")) {
+                    // on is an alias for '1'
+                    Ok(FeatureTagValue{ tag: tag, value: 1 })
+                } else if let Ok(_) = input.try(|input| input.expect_ident_matching("off")) {
+                    // off is an alias for '0'
+                    Ok(FeatureTagValue{ tag: tag, value: 0 })
+                } else {
+                    // empty value is an alias for '1'
+                    Ok(FeatureTagValue{ tag:tag, value: 1 })
+                }
+            }
+        }
+    }
+
+    #[inline]
+    pub fn get_initial_value() -> computed_value::T {
+        computed_value::T::Normal
+    }
+
+    /// normal | <feature-tag-value>#
+    pub fn parse(_context: &ParserContext, input: &mut Parser) -> Result<SpecifiedValue, ()> {
+        if input.try(|input| input.expect_ident_matching("normal")).is_ok() {
+            Ok(computed_value::T::Normal)
+        } else {
+            input.parse_comma_separated(computed_value::FeatureTagValue::parse)
+                 .map(computed_value::T::Computed)
```

This should return an error if the length is `0`